third_party/dependency_analyzer/src/test/io/bazel/rulesscala/dependencyanalyzer/ScalaVersionTest.scala

```diff
@@ -0,0 +1,136 @@
+package third_party.dependency_analyzer.src.test.io.bazel.rulesscala.dependencyanalyzer
+
+import org.scalatest._
+import third_party.dependency_analyzer.src.main.io.bazel.rulesscala.dependencyanalyzer.ScalaVersion
+
+class ScalaVersionTest extends FunSuite {
```

Comment:
Big big like on these tests! Well done :)

Can you add a test for the inclusive min max case?
The con for it is that it will duplicate a bit the prod code of getting the current version but I think the confidence we will get from someone accidentally changing the `>=` to `>` or `<=` to `<` is big

Reply:
Due to the way macros are implemented I don't think it's possible, since we need to use inlineable strings (which current version is not, it seems). Unless we wanted to do templating or something but that shouldn't be the case.

FWIW, hopefully anything that uses macros will be tested, and the test version checks are straightforward, so if there is a bug in the prod code it should likely be caught by unit tests when run against all versions.

Reply:
Done